src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/MemberExpression.cs

```diff
@@ -150,6 +150,7 @@ public static MemberExpression Field(Expression? expression, FieldInfo field)
         /// <param name="expression">The containing object of the field.  This can be null for static fields.</param>
         /// <param name="fieldName">The field to be accessed.</param>
         /// <returns>The created <see cref="MemberExpression"/>.</returns>
+        [RequiresUnreferencedCode(ExpressionRequiresUnreferencedCode)]
```

Comment:
I haven't realizes that linker has intrinsic handling for some of the `Expression.` static creation methods. For example field (and property) are here: https://github.com/mono/linker/blob/319ed02ed9a25ec3800f723ff72a1eb92ef050c7/src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs#L722

I don't think there's anything to be done here - linker will not report the warning if the intrinsic handling kicks in - we should leave the method annotated like this for sure.